syntax = "proto3";
package installed_micro_app;




import "google/protobuf/struct.proto";

/*
SearchInstalledMicroAppWithUniqKey请求
*/
message SearchInstalledMicroAppWithUniqKeyRequest {
message Query {
    /*
    app name
    */
    string name = 1 ;
    /*
    appid
    */
    string app_id = 2 ;
    /*
    homepage
    */
    string homepage = 3 ;
}
        
    
    /*
    页码
    */
    int32 page = 1 ;
    /*
    每页大小
    */
    int32 page_size = 2 ;
    /*
    查询条件
    */
    Query query = 3 ;
}

/*
SearchInstalledMicroAppWithUniqKey返回
*/
message SearchInstalledMicroAppWithUniqKeyResponse {
message List {
message Icons {
    /*
    图标url
    */
    string large = 1 ;
}
        
message ClonedFrom {
    /*
    克隆的appId
    */
    string appId = 1 ;
    /*
    克隆的版本号
    */
    string version = 2 ;
    /*
    克隆的app名称
    */
    string name = 3 ;
}
        
message MenuIcon {
    /*
    图标库
    */
    string lib = 1 ;
    /*
    Antd 图标类型
    */
    string type = 2 ;
    /*
    Antd 图标主题
    */
    string theme = 3 ;
    /*
    FA 或 EasyOps 图标
    */
    string icon = 4 ;
    /*
    FA 图标前缀
    */
    string prefix = 5 ;
    /*
    EasyOps 图标分类
    */
    string category = 6 ;
}
        
    
    /*
    cmdb唯一标识, 业务逻辑请使用appId作为唯一标识
    */
    string instanceId = 1 ;
    /*
    小产品名称
    */
    string name = 2 ;
    /*
    小产品id
    */
    string appId = 3 ;
    /*
    图标url
    */
    Icons icons = 4 ;
    /*
    小产品配置
    */
    string storyboardJson = 5 ;
    /*
    标签
    */
    repeated string tags = 6 ;
    /*
    NA程序包当前版本
    */
    string currentVersion = 7 ;
    /*
    小产品在应用商店的版本号
    */
    string appVersion = 8 ;
    /*
    安装状态， ok-成功, running-正在安装
    */
    string installStatus = 9 ;
    /*
    小产品首页
    */
    string homepage = 10 ;
    /*
    表示该应用是内部的，不出现在 launchpad 和 app store 中
    */
    string internal = 11 ;
    /*
    私有安装应用, true or false
    */
    string private = 12 ;
    /*
    克隆对象
    */
    ClonedFrom clonedFrom = 13 ;
    /*
    作者
    */
    string owner = 14 ;
    /*
    readme
    */
    string readme = 15 ;
    /*
    状态
    */
    string status = 16 ;
    /*
    创建时间
    */
    string ctime = 17 ;
    /*
    修改时间
    */
    string mtime = 18 ;
    /*
    关联程序包名称
    */
    string pkgName = 19 ;
    /*
    图标背景
    */
    string iconBackground = 20 ;
    /*
    菜单中显示的图标
    */
    MenuIcon menuIcon = 21 ;
    /*
    默认配置
    */
    google.protobuf.Struct defaultConfig = 22 ;
    /*
    用户配置
    */
    google.protobuf.Struct userConfig = 23 ;
}
        
    
    /*
    返回总数
    */
    int32 total = 1 ;
    /*
    页数
    */
    int32 page = 2 ;
    /*
    该页大小
    */
    int32 page_size = 3 ;
    /*
    数据列表
    */
    repeated List list = 4 ;
}

/*
SearchInstalledMicroAppWithUniqKeyApi返回
*/
message SearchInstalledMicroAppWithUniqKeyResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    installed_micro_app.SearchInstalledMicroAppWithUniqKeyResponse data = 4 ;
}

